recipes/capnproto/all/conanfile.py

```diff
@@ -0,0 +1,183 @@
+from conans import ConanFile, CMake, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import glob
+import os
+
+
+class CapnprotoConan(ConanFile):
+    name = "capnproto"
+    description = "Cap'n Proto serialization/RPC system."
+    license = "MIT"
+    topics = ("conan", "capnproto", "serialization", "rpc")
+    homepage = "https://capnproto.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ("CMakeLists.txt", "patches/**")
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False],
+        "with_zlib": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True,
+        "with_zlib": True
+    }
+
+    _cmake = None
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            del self.options.with_openssl
```

Comment:
Maybe not delete the option, but throw when the option is enabled on Windows?
If you say it could be possible to re-enable it in the future.
(For now maybe disable it on Windows, enable it on other platforms with a FIXME)

Reply:
let's just add a small patch to build `kj-tls` in CMake, it avoids complex options handling depending on platform and lib version.

Reply:
I would completely switch over to cmake in next releases, that would avoid the simple but huge autotools patch.

Reply:
Well basically, the first proposition for this PR was to use CMake only, but library author doesn't trust it: https://github.com/conan-io/conan-center-index/pull/1489#issuecomment-621495851